components/family-studies/api/src/main/java/org/phenotips/studies/family/internal/ValidationImpl.java

```diff
@@ -211,4 +236,27 @@ private boolean hasPatientAccess(Patient patient, AccessLevel accessLevel, User
         AccessLevel patientAccessLevel = patientAccess.getAccessLevel(user.getProfileDocument());
         return patientAccessLevel.compareTo(accessLevel) >= 0;
     }
+
+    @Override
+    public boolean hasAccess(DocumentReference document, String permissions)
+    {
+        XWikiDocument xWikiDoc = null;
+        try {
+            xWikiDoc = this.familyUtils.getDoc(document);
+        } catch (XWikiException e) {
+            this.logger.error("Error retrieving family document for family [{}]: [{}]",
+                document.getName(), e.getMessage());
+        }
+
+        XWikiContext context = this.provider.get();
+        XWiki wiki = context.getWiki();
+
+        try {
+            return wiki.checkAccess(permissions, xWikiDoc, context);
```

Comment:
Don't use this method, it will mess the response if the check fails, since it will redirect to the login page. You can use one of:
* `wiki.getRightService().hasAccessLevel(...)`
* `org.xwiki.security.authorization.AuthorizationManager#hasAccess` (which you can `@Inject`)
* `org.phenotips.security.authorization.AuthorizationService.hasAccess()` (which you can also `@Inject`)

The last one is the best.